src/pass/lift_if_then_else.cc

```diff
@@ -0,0 +1,422 @@
+      for_tracking_map_[for_stmt.get()].at(updated_for_idx);
+    auto generated_for_pair = RemoveIf(updated_for_node, new_if);
+    const Stmt& then_for = generated_for_pair.first;
+    const Stmt& else_for = generated_for_pair.second;;
+    for_tracking_map_[for_stmt.get()].at(updated_for_idx) = then_for;
+
+    if (else_for.get()) {
+      for_tracking_map_[for_stmt.get()].push_back(else_for);
+    }
+
+    const IfThenElse* new_if_node = new_if.as<IfThenElse>();
+    CHECK(new_if_node);
+    new_if = IfThenElse::make(new_if_node->condition, then_for, else_for);
+    if (i < if2for_map_[if_stmt.get()].size() - 1) {
+      const Stmt& original_next_for = if2for_map_[if_stmt.get()].at(i + 1);
+      const Stmt& actual_next_for =
+        for_tracking_map_[original_next_for.get()].at(updated_for_idx);
+      Stmt update_for_stmt = update_for(actual_next_for, new_if);
+
+      for_tracking_map_[original_next_for.get()].
+        at(updated_for_idx) = update_for_stmt;
+    }
+  }
+  return new_if;
+}
+
+// Mutate For nodes in post order DFS manner.
+Stmt IfThenElseLHoist::PostOrderMutate(const Stmt& stmt) {
+  PackedFunc replace_top_for = PackedFunc(
+    [&](TVMArgs args, TVMRetValue *ret){
+      const NodeRef& current_for = args[0];
+      const For* for_node = current_for.as<For>();
+      if (for_node) {
+        if (top_for_var_map_.count(for_node->loop_var.get())) {
+          std::vector<Stmt> new_if_list;
+          for (const Stmt& if_stmt :
+            top_for_var_map_[for_node->loop_var.get()]) {
+            new_if_list.emplace_back(HoistIf(if_stmt));
+          }
+
+          const IfThenElse* next_if_node;
+          const IfThenElse* current_if_node =
+            new_if_list.back().as<IfThenElse>();
+          Stmt new_for = Stmt();
+          for (size_t i = new_if_list.size() - 1; i > 0; --i) {
+            CHECK(current_if_node);
+            const Stmt current_if_stmt =
+              IfThenElse::make(current_if_node->condition,
+                               current_if_node->then_case,
+                               current_if_node->else_case);
+            next_if_node = new_if_list[i - 1].as<IfThenElse>();
+            CHECK(next_if_node);
+            new_for = IfThenElse::make(next_if_node->condition, current_if_stmt,
+                                       next_if_node->else_case);
+            current_if_node = new_for.as<IfThenElse>();
+          }
+
+          if (!new_for.get()) {
+            const IfThenElse* first_if_node = new_if_list[0].as<IfThenElse>();
+            CHECK(first_if_node);
+            new_for = IfThenElse::make(first_if_node->condition,
+                                       first_if_node->then_case,
+                                       first_if_node->else_case);
+          }
+          *ret = new_for;
+        }
+      }
+    });
+  return IRTransform(stmt, nullptr, replace_top_for, {Expr("For")});
+}
+
+Stmt HoistIfThenElse(Stmt stmt) {
+  return IfThenElseLHoist().VisitAndMutate(stmt);
```

Comment:
What does 'L' stand for in `IfThenElseLHoist`?

Reply:
This is a typo.